NMDSdevopsServiceAdm/DataEngineering
# Add error handling postcodes

# Description
https://trello.com/c/X2rf20h6/179-epic-4-add-error-handling-for-postcodes-must

# How to test
Outside of the regular unit tests, pay attention to the changes to the main method as I had trouble doing anything more complicated than this, so see if this will suffice.
I also need to test this in Glue to make sure it errors out as expected, **but this is known about and we are expecting an error at this stage**

## Before: 
![Screenshot 2024-03-12 at 14 23 36](https://github.com/NMDSdevopsServiceAdm/DataEngineering/assets/108728588/fbc758ba-275b-47b8-81c4-213004b262a6)

## After:
![Screenshot 2024-03-13 at 14 04 10](https://github.com/NMDSdevopsServiceAdm/DataEngineering/assets/108728588/4bd8f2a2-2002-431f-99aa-c9de571a6c2a)
Successful run

# Developer checklist
